xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAE.cpp

```diff
@@ -163,12 +163,30 @@ void CEngineStats::SetDSP(bool state)
   m_hasDSP = state;
 }
 
+void CEngineStats::SetCurrentSinkFormat(AEAudioFormat &SinkFormat)
+{
+  CSingleLock lock(m_lock);
+  m_sinkFormat = SinkFormat;
+}
+
 bool CEngineStats::HasDSP()
 {
   CSingleLock lock(m_lock);
   return m_hasDSP;
 }
 
+bool CEngineStats::GetCurrentSinkFormat(AEAudioFormat &SinkFormat)
+{
+  CSingleLock lock(m_lock);
+  if(m_sinkFormat.m_dataFormat == AE_FMT_INVALID)
```

Comment:
why this? imo it is much simpler just to return the format. we have AE_FMT_INVALID, hence you don't need a bool to determine if the format is valid.